connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -1264,6 +1272,28 @@ public void testMetricsGroup() {
         assertEquals(30, metrics.currentMetricValueAsDouble(group1.metricGroup(), "put-batch-max-time-ms"), 0.001d);
     }
 
+    @Test
+    public void testHeaders() throws Exception {
+        createTask(initialState);
+
+        expectInitializeTask();
+        expectPollInitialAssignment();
+
+        expectConsumerPoll(1, HEADERS);
+        expectConversionAndTransformation(1, null, HEADERS);
+        sinkTask.put(EasyMock.<Collection<SinkRecord>>anyObject());
+        EasyMock.expectLastCall();
+
+        PowerMock.replayAll();
+
+        workerTask.initialize(TASK_CONFIG);
+        workerTask.initializeAndStart();
+        workerTask.iteration(); // iter 1 -- initial assignment
+        workerTask.iteration(); // iter 2 -- deliver 1 record
+
+        PowerMock.verifyAll();
```

Comment:
These tests just seem to validate that your new method is called, but not much else. Perhaps provide an implementation that actually does something with the headers, and then test that.

Reply:
I agree. It'd probably be good to use a converter implementation that fails if the non-header form of the methods are called, but does something if the header form of the methods are called.

Reply:
I've added `TestConverterWithHeaders` and sink/source tests showing its usage.